### DISCUSSION TITLE
Using $persist with x-show

### DISCUSSION THREAD

**Discussion by ericp-mrel** [alpinejs/alpine#2163] [1. Help]
I'm trying to use the $persist plugin along with the x-show in order to show a splash screen intro animation upon the user's first visit only. But, it seems this component flashes on the screen (after first page load) for a second until the $persist is fully evaluated and then finally hides the component.

And on slower connections (like fast 3G network mode in Chrome), the video will actually fully play even if the local storage value is already set to `true`. I also assumed adding the x-cloak would've prevented this type of a situation

Here is the code I'm trying to use:
```html
<div class="fixed inset-0 w-full h-full bg-white grid place-items-center" style="z-index: 999999;" x-cloak x-show="!viewed" x-data="{viewed: $persist(false).as('viewed-splash-intro')}" x-transition.opacity>
	<video width="1280" height="720" muted autoplay @ended="viewed = true">
		<source src="/assets/video.mp4" type="video/mp4">
		Sorry, your browser doesn't support embedded videos.
	</video>
</div>
```
I've even tried doing something like this, but this still would play the video animation on slow connection.
```html
<div class="fixed inset-0 w-full h-full bg-white grid place-items-center" style="z-index: 999999;" x-cloak x-show="initialized && !viewed" x-init="initialized = true" x-data="{initialized: false, viewed: $persist(false).as('viewed-splash-intro')}">
    <video width="1280" height="720" muted autoplay @ended="viewed = true">
        <source src="/assets/video.mp4" type="video/mp4">
        Sorry, your browser doesn't support embedded videos.
    </video>
</div>
```
In the end, it seems like the only way I could really get it to work how I wanted to was by adding a `display: none` inline style:
```html
<div class="fixed inset-0 w-full h-full bg-white grid place-items-center" style="z-index: 999999; display: none" x-cloak x-show="!viewed" x-data="{viewed: $persist(false).as('viewed-splash-intro')}">
	<video width="1280" height="720" muted autoplay @ended="viewed = true">
		<source src="/assets/video.mp4" type="video/mp4">
		Sorry, your browser doesn't support embedded videos.
	</video>
</div>
```

Not sure if anyone knows of a better way to accomplish what I want? Or if I was just missing something in the original attempt.

**Comment by johndwells** [comment]
I don't have a better solution, but I can confirm I'm seeing this as well.

I wonder if this scenario is an argument for restoring the removal of `x-cloak` via `nextTick` instead of a microtask: https://github.com/alpinejs/alpine/pull/1570/commits/837397a06cbf75d663e367145e05cc1a9d1a0388 which was raised during this discussion: https://github.com/alpinejs/alpine/discussions/1554

Or perhaps your workaround (to use `display: none` & `x-show` instead of `x-cloak`) becomes the documented alternative for such a situation?

**Reply by SimoTod** [reply]
Can you create a codepen to replicate. It seems to work correctly for me.

**Reply by ericp-mrel** [reply]
@SimoTod 

Here's a codepen of my original code attempt: https://codepen.io/ericp_mrel/pen/oNebxmG
And here's a fullscreen preview: https://codepen.io/ericp_mrel/full/oNebxmG

First, let the video finish playing and once it has ended it will hide the overlay and show the blue screen. And then if you reload the page you can see it flash a white screen (which is the overlay video) for a brief second which should not happen. The reason the Alpine component should never be shown in the first place (even for the brief flash is because x-cloak should hide it until it has been initialized, x-show should then be initialized with the stored value from local storage which should evaluate to true on page loads after the video has been played once on initial visit. But it's like there is a few milliseconds between the time the x-cloak is removed and x-show is set to true from localstorage.

**Reply by SimoTod** [reply]
Thanks, I changed the background to `bg-red-500` but I still see the white for a millisecond (opposite to red) so I believe it's just codepen loading the iframe.

**Reply by ericp-mrel** [reply]
Ah, yeah it's possible. I wonder if the issue I was experiencing from my original post only happens when using the NPM version? Because, it seems to work fine on codepen using the CDN build.

**Reply by SimoTod** [reply]
They should behave the same but obviously it's hard to tell without seeing the issue. You mentioned a slow connection, was your css rule for x-cloak bundled with other files and download as an external script? To guarantee that those divs won't show before Alpine kicks in, that rule must be added to a style tag on your page so it's always available.
